syne_tune/optimizer/schedulers/transfer_learning/rush.py

```diff
@@ -0,0 +1,151 @@
+
+
+class RUSHScheduler(TransferLearningScheduler):
+    def __init__(
+            self,
+            config_space: Dict,
+            transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+            metric: str,
+            **kwargs
+    ) -> None:
+        """
+        A transfer learning variation of Hyperband which uses previously well-performing hyperparameter configurations
+        as an initialization. Additionally, the performance of these initial configuration at the different rung levels
+        serve as a hurdle for other sampled configurations. No candidate configuration which is performing worse
+        than the best initial candidate, will be promoted.
+        Simple baseline that computes a bounding-box of the best candidate found in previous tasks to restrict the
+         search space to only good candidates. The bounding-box is obtained by restricting to the min-max of best
+         numerical hyperparameters and restricting to the set of best candidates on categorical parameters.
+
+        Reference: A resource-efficient method for repeated HPO and NAS.
+        Giovanni Zappella, David Salinas, Cédric Archambeau. AutoML workshop @ ICML 2021.
+
+        :param config_space: Configuration space for trial evaluation function.
+        :param metric: objective name to optimize, must be present in transfer learning evaluations.
+        :param transfer_learning_evaluations: dictionary from task name to offline evaluations.
+        """
+        super().__init__(config_space=config_space,
+                         transfer_learning_evaluations=transfer_learning_evaluations,
+                         metric_names=[metric])
+        points_to_evaluate = RUSHScheduler._determine_baseline_configurations(
+            config_space=config_space,
+            transfer_learning_evaluations=transfer_learning_evaluations,
+            metric=metric,
+            mode=kwargs.get('mode', 'min')
+        )
+        if 'points_to_evaluate' in kwargs:
+            points_to_evaluate += kwargs['points_to_evaluate']
+            points_to_evaluate = [dict(s) for s in set(frozenset(p.items()) for p in points_to_evaluate)]
+        kwargs['points_to_evaluate'] = points_to_evaluate
+
+        self._hyperband_scheduler = HyperbandScheduler(config_space, metric=metric, **kwargs)
+        self._num_init_configs = len(points_to_evaluate)
+        self._thresholds = dict()  # thresholds at different resource levels that must be met
+
+    @staticmethod
+    def _determine_baseline_configurations(config_space: Dict,
+                                           transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+                                           metric: str,
+                                           mode: str) -> List[Dict]:
+        argbest, best = (np.argmin, np.min) if mode == 'min' else (np.argmax, np.max)
```

Comment:
Could you add a comment such as `# compute the best hyperparameter on each task`? (Or possibly we could also rename the method to something like `_compute_best_hyperparameter_per_task`)